test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlClientInstrumentationOptionsTests.cs

```diff
@@ -102,5 +104,106 @@ static SqlClientInstrumentationOptionsTests()
             Assert.Equal(expectedInstanceName, activity.GetTagValue(SemanticConventions.AttributeDbMsSqlInstanceName));
             Assert.Equal(expectedPort, activity.GetTagValue(SemanticConventions.AttributeNetPeerPort));
         }
+
+        // Tests for newer HTTP v1.21.0 Semantic Conventions.
+        // This test method can replace the other when this library is GA.
+        [Theory]
+        [InlineData(true, "localhost", "localhost", null, null, null)]
+        [InlineData(true, "127.0.0.1,1433", null, "127.0.0.1", null, null)]
+        [InlineData(true, "127.0.0.1,1434", null, "127.0.0.1", null, "1434")]
+        [InlineData(true, "127.0.0.1\\instanceName, 1818", null, "127.0.0.1", "instanceName", "1818")]
+        [InlineData(false, "localhost", "localhost", null, null, null)]
+        public void SqlClientInstrumentationOptions_EnableConnectionLevelAttributes_New(
+            bool enableConnectionLevelAttributes,
+            string dataSource,
+            string expectedServerHostName,
+            string expectedServerIpAddress,
+            string expectedInstanceName,
+            string expectedPort)
+        {
+            var source = new ActivitySource("sql-client-instrumentation");
+            var activity = source.StartActivity("Test Sql Activity");
+
+            var configuration = new ConfigurationBuilder()
+                .AddInMemoryCollection(new Dictionary<string, string> { [SemanticConventionOptInKeyName] = "http" })
+                .Build();
+
+            var options = new SqlClientInstrumentationOptions(configuration)
+            {
+                EnableConnectionLevelAttributes = enableConnectionLevelAttributes,
+            };
+            options.AddConnectionLevelDetailsToActivity(dataSource, activity);
+
+            if (!enableConnectionLevelAttributes)
+            {
+                Assert.Equal(expectedServerHostName, activity.GetTagValue(SemanticConventions.AttributePeerService));
+            }
+            else
+            {
+                Assert.Equal(expectedServerHostName, activity.GetTagValue(SemanticConventions.AttributeServerAddress));
+            }
+
+            Assert.Equal(expectedServerIpAddress, activity.GetTagValue(SemanticConventions.AttributeServerSocketAddress));
+            Assert.Equal(expectedInstanceName, activity.GetTagValue(SemanticConventions.AttributeDbMsSqlInstanceName));
+            Assert.Equal(expectedPort, activity.GetTagValue(SemanticConventions.AttributeServerPort));
+        }
+
+        // Tests for newer HTTP v1.21.0 Semantic Conventions and older attributes.
```

Comment:
It seems we still have confusion here, what exactly is "HTTP v1.21.0 Semantic Conventions", @TimothyMothra could you put a concrete link?

Reply:
Corrected this. New text:

```
// Tests for v1.21.0 Semantic Conventions for database client calls.
// see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/trace/semantic_conventions/database.md
// This test emits both the new and older attributes.
// This test method can be deleted when this library is GA.
```